Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "URDFPrefabMaker.h"
+#include "CollidersMaker.h"
+#include "PrefabMakerUtils.h"
+#include <API/EditorAssetSystemAPI.h>
+#include <AzCore/IO/FileIO.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <AzToolsFramework/Prefab/PrefabLoaderInterface.h>
+#include <AzToolsFramework/Prefab/PrefabSystemComponentInterface.h>
+#include <AzToolsFramework/ToolsComponents/GenericComponentWrapper.h>
+#include <AzToolsFramework/ToolsComponents/TransformComponent.h>
+#include <ROS2/Frame/ROS2FrameComponent.h>
+#include <ROS2/ROS2GemUtilities.h>
+#include <ROS2/Spawner/SpawnerBus.h>
+#include <RobotControl/ROS2RobotControlComponent.h>
+#include <RobotImporter/Utils/RobotImporterUtils.h>
+
+namespace ROS2
+{
+    URDFPrefabMaker::URDFPrefabMaker(
+        const AZStd::string& modelFilePath,
+        urdf::ModelInterfaceSharedPtr model,
+        AZStd::string prefabPath,
+        const AZStd::shared_ptr<Utils::UrdfAssetMap> urdfAssetsMapping)
+        : m_model(model)
+        , m_visualsMaker(model->materials_, urdfAssetsMapping)
+        , m_collidersMaker(urdfAssetsMapping)
+        , m_prefabPath(std::move(prefabPath))
+        , m_urdfAssetsMapping(urdfAssetsMapping)
+    {
+        AZ_Assert(!m_prefabPath.empty(), "Prefab path is empty");
+        AZ_Assert(m_model, "Model is nullptr");
+    }
+
+    void URDFPrefabMaker::LoadURDF(BuildReadyCallback buildReadyCb)
+    {
+        m_notifyBuildReadyCb = buildReadyCb;
+
+        // Request the build of collider meshes by constructing .assetinfo files.
+        BuildAssetsForLink(m_model->root_link_);
+
+        // Wait for all collider meshes to be ready
```

Comment:
This comment doesn't match the function call. Is this correct?

Reply:
Done